recipes-extended/onie-tools/onie-tools/backup.sh

```diff
@@ -153,70 +193,125 @@ backup_systemd_state() {
 				# nothing to do if still enabled
 				[ -z "$enabled" ] || continue
 
-				[ -n "$DEBUG" ] && echo "DEBUG: service disabled by user: $service"
+				[ -n "$DEBUG" ] && echo "DEBUG: service disabled by user: $service" >&2
 				disabled_services="$disabled_services ${service%.service}"
 				;;
 		esac
 	done
 
-	# collect explicitly enabled services
-	for service in $service_links; do
-		case "$service" in
-			*.service)
-				service_file=$(readlink $1/etc/systemd/system/multi-user.target.wants/$service)
-				base=$(basename $service_file)
-				preset=$(grep "$base" $1/lib/systemd/system-preset/*.preset | cut -d ':' -f 2 | cut -d ' ' -f 1)
+	echo "$disabled_services"
+}
 
-				# check only services disabled by default
-				[ "$preset" != "enable" ] || continue
+copy_files() {
+	local files=$1
+	local src=${2%/}
+	local dst=${3%/}
+	local file
 
-				[ -n "$DEBUG" ] && echo "DEBUG: service enabled by user: $service"
-				enabled_services="$enabled_services ${service%.service}"
-				;;
-		esac
+	if [ -n "$DRYRUN" ]; then
+		echo "Files to backup:"
+		[ -z "$files" ] || echo "$files"
+		echo ""
+		return 0
+	fi
+
+	for file in $files; do
+		if [ "$(dirname $file)" != "/" ]; then
+			cp_path_with_attr "$(dirname $src/$file)" "$(dirname $dst/$file)"
+		fi
+		cp -a "$src/$file" "$(dirname $dst/$file)"
 	done
+}
 
-	echo "$disabled_services" > $2/.SERVICES_DISABLED
-	echo "$enabled_services" > $2/.SERVICES_ENABLED
+backup_systemd_state() {
+	local enabled_services disabled_services
+
+	enabled_services=$(collect_enabled_services $1)
+	disabled_services=$(collect_disabled_services $1)
+
+	if [ -n "$DRYRUN" ]; then
+		echo "Services to enable:"
```

Comment:
As a user, I would wonder what this actually does (if I create a backup while, where do I find the backup for the service state?). Maybe the usage text or the man page could elaborate (list stored in `.SERVICES_ENABLED/.SERVICES_DISABLED`, then enabled/disabled via systemctl).